packages/autometrics-lib/src/wrappers.ts

```diff
@@ -244,56 +207,60 @@ export function autometrics<F extends FunctionSig>(
   };
 }
 
-function isPromise<T extends Promise<void>>(val: unknown): val is T {
-  return (
-    typeof val === "object" &&
-    "then" in val &&
-    typeof val.then === "function" &&
-    "catch" in val &&
-    typeof val.catch === "function"
-  );
-}
-
-// HACK: this entire function is a hacky way to acquire the module name
-// for a given function e.g.: dist/index.js
-function getModulePath(): string | undefined {
-  Error.stackTraceLimit = 5;
-  const stack = new Error()?.stack?.split("\n");
-
-  let rootDir: string;
-
-  if (typeof process === "object") {
-    // HACK: this assumes the entire app was run from the root directory of the project
-    rootDir = process.cwd();
-    //@ts-ignore
-  } else if (typeof Deno === "object") {
-    //@ts-ignore
-    rootDir = Deno.cwd();
-  } else {
-    rootDir = "";
-  }
-
-  if (!stack) {
-    return;
-  }
-
-  /**
-   *
-   * 0: Error
-   * 1: at getModulePath() ...
-   * 2: at autometrics() ...
-   * 3: at ... -> 4th line is always the original caller
-   */
-  const originalCaller = 3 as const;
-
-  // The last element in this array will have the full path
-  const fullPath = stack[originalCaller].split(" ").pop();
-
-  // We split away everything up to the root directory of the project
-  let modulePath = fullPath.replace(rootDir, "");
-
-  // We split away the line and column numbers index.js:14:6
-  modulePath = modulePath.substring(0, modulePath.indexOf(":"));
+/**
+ * Autometrics decorator that can be applied to either a class or class method
+ * that automatically instruments methods with OpenTelemetry-compatible metrics.
+ * Hover over the method to get the links for generated queries (if you have the
+ * language service plugin installed).
+ *
+ * Optionally, you can pass in an {@link AutometricsOptions} object to configure
+ * the decorator.
+ * @param autometricsOptions
+ * @example
+ * ```typescript
+ * import {
+ *   Autometrics,
+ *   AutometricsOptions,
+ *   Objective,
+ *   ObjectivePercentile,
+ *   ObjectiveLatency,
+ * } from "autometrics";
+ *
+ * const objective: Objective = {
+ *   successRate: ObjectivePercentile.P99_9,
+ *   latency: [ObjectiveLatency.Ms250, ObjectivePercentile.P99],
+ *   name: "foo",
+ * };
+ *
+ * const autometricsOptions: AutometricsOptions = {
+ *   functionName: "FooBar",
+ *   moduleName: "FooModule",
+ *   objective,
+ *   trackConcurrency: true,
+ * };
+ *
+ * class Foo {
+ *   \@Autometrics(autometricsOptions)
```

Comment:
small typo correction in the example

```suggestion
 *   @Autometrics(autometricsOptions)
```

Reply:
This is intentional: I'll add a clarifying comment to the file. The issue is that the code block doesn't properly escape the `@` sign and renders it as:
```typescript
// ...
*@Autometrics*  
(autometricsOptions)
```

I found multiple issues with this problem, but the sign cannot be properly escaped without hacky solutions; hence why I opted for the backslash here which'll render it proper (albeit with the `\` still visible).

Reply:
ah interesting ok! JSDoc picks it up it seems